index.js

```diff
@@ -1,97 +1,215 @@
+            const message = `:globe_with_meridians: ${urlName} (${urlValue}) is up and running!`;
+            send(message);
+            console.log(`${getFormattedDate(responseDate)} - ${message}`);
+        }
+    }
+
+    urlContainer[urlName] = UrlStatus.Up;
+}
+
+function handleLocalError(urlName, responseDate) {
+    console.log('Local error, couldn\'t reach endpoint.');
+
+    if (urlErrorContainer[urlName] === undefined) {
+        urlErrorContainer[urlName] = responseDate;
+    }
+}
+
+function handleDownStatus(urlName, params, responseDate) {
+    const trailingMentions = buildMentionsForDownStatus(params.mentionsWhenDown);
+    const message = `:warning: Oh no! It looks like ${urlName} (${params.url}) is down.\n${trailingMentions}`;
+    console.log(`${getFormattedDate(responseDate)} - ${message}`);
+
+    if (urlContainer[urlName] === undefined) {
+        send("First check of this session... Endpoint didn't respond :confused:");
+        urlContainer[urlName] = UrlStatus.Pending;
+        return;
+    }
+
+    if (urlErrorContainer[urlName] !== undefined) {
+        restoreLocalErrorAndSend(responseDate, urlName);
+    }
+
+    if (urlContainer[urlName] === UrlStatus.Up) {
+        // First fail! Try again, maybe this was temporary.
+        urlContainer[urlName] = UrlStatus.Pending;
+    } else if (urlContainer[urlName] === UrlStatus.Pending) {
+        // Two fails in a row! Send message, the endpoint is down.
+        urlContainer[urlName] = UrlStatus.Down;
+        send(message);
+    }
+}
+
+function buildMentionsForDownStatus(mentions) {
+    if (mentions === undefined) return '';
+
+    return mentions.reduce((total, mention, index) => {
+        if (mention.user === undefined || mention.group === undefined) return total;
+
+        const leadingText = (index === 0) ? 'cc: ' : ', ';
+        const trailingText = (mention.user) ? `<@${mention.id}>` : `<!subteam^${mention.id}>`;
+
+        return `${total}${leadingText}${trailingText}`
+    }, '');
 }
 
 /* Send message to slack incoming webhook */
 function send(text) {
-    webhook.send(text, function (err, res) {
-        if (err) {
-            console.log('Error:', err);
-        } else {
-            console.log('Message sent: ', res);
+    console.log(webhookUrl);
+    request.post(
+        webhookUrl,
+        {
+            json: { text: text }
+        },
+        function (err, res) {
+            if (err) {
+                console.log(`Error: ${err}`)
+            } else {
+                console.log(`Message sent: ${res}`)
+            }
         }
-    });
+    );
+}
+
+function restoreLocalErrorAndSend(currentDate, urlName) {
+    const localErrorSince = urlErrorContainer[urlName];
+
+    if (localErrorSince === undefined) return;
+
+    const durationInLocalError = moment.duration(moment(currentDate).diff(moment(localErrorSince)))
+    const hasBeenMoreThanOneMinute = durationInLocalError.asSeconds() > 60
```

Comment:
I'd change from 60 to a configurable threshold, default to 60 (or 60000 to be in line with the other configuration).

It would make code as simple but allows for configurable duration